distsql/select_result.go

```diff
@@ -145,12 +147,35 @@ func (r *selectResult) NextRaw(ctx context.Context) (data []byte, err error) {
 // Next reads data to the chunk.
 func (r *selectResult) Next(ctx context.Context, chk *chunk.Chunk) error {
 	chk.Reset()
+	// Check the returned data is default/arrow format.
+	if r.selectResp == nil || (len(r.selectResp.RowBatchData) == 0 && r.respChkIdx == len(r.selectResp.Chunks)) {
+		err := r.getSelectResp()
+		if err != nil || r.selectResp == nil {
+			return err
+		}
+		if len(r.selectResp.RowBatchData) == 0 {
+			r.encodeType = tipb.EncodeType_TypeDefault
```

Comment:
IMHO, we only need to do this check when the `Next()` function is called the first time. All the cop responses should have the same decode method. Correct me if I'm wrong.

Reply:
You are right.

Reply:
Now the check is temporary ad-hoc. The next pr will add a field `response_format` in response, and use the field to decide the decode format. Then i will remove the check in next pr. It is OK?